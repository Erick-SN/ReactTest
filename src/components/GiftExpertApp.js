import { useState } from 'react';
import AddCategory from './AddCategory';

const GiftExpertApp = () => {
  const [categories, setCategories] = useState(['DBZ', 'HXH', 'OP']);
  const handleAddCategory = (e) => {
    e.preventDefault();
    setCategories([...categories, 'NNT']);
  };

  return (
    <>
      <h2>Encuentra tus gifts preferidos!</h2>
      <AddCategory />
      <hr />
      <button onClick={handleAddCategory}>Agregar</button>
      <ol>
        {categories.map((category) => (
          <li key={category}>{category}</li>
        ))}
      </ol>
    </>
  );
};

export default GiftExpertApp;
