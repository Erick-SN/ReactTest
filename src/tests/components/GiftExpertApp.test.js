import { shallow } from 'enzyme';
import GiftExpertApp from '../components/GiftExpertApp';

describe('Pruebas en el componente < GiftExpertApp/>', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(GiftExpertApp);
  });
  test('Prueba de renderizado correcto', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
