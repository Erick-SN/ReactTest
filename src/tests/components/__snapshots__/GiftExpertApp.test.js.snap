// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente < GiftExpertApp/> Prueba de renderizado correcto 1`] = `
<Fragment>
  <h2>
    Encuentra tus gifts preferidos!
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiftGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
